src/pages/index.js

```diff
@@ -112,22 +181,77 @@ function closeModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModalProfile);
+  setButtonText(getSubmitButton(evt), true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModalProfile);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(getSubmitButton(evt), false);
+    });
 }
 
 function handleAddPostSubmit(evt) {
   evt.preventDefault();
+  const form = evt.target;
   const inputValues = {
-    name: addPostCaptionInput.value,
-    link: addPostLinkInput.value,
+    name: form.elements["add-post-caption-input"].value,
+    link: form.elements["add-post-link-input"].value,
   };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  closeModal(addPostModal);
-  disabledButton(addPostSubmitButton, settings);
+  setButtonText(getSubmitButton(evt), true);
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      closeModal(addPostModal);
+      disabledButton(addPostSubmitButton, settings);
```

Comment:
* There is an error when I add a new card https://snipboard.io/mFV6M5.jpg 
* Please, **export** the `disabledButton` function from `validation.js`, to reuse it in `index.js` for disabling